contract_compassion/model/contracts.py

```diff
@@ -0,0 +1,646 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2014 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Cyril Sester, Emanuel Cino
+#
+#    The licence is in the file __openerp__.py
+#
+##############################################################################
+
+from openerp.osv import orm, fields
+from openerp import netsvc
+from openerp.tools.translate import _
+from openerp.tools import DEFAULT_SERVER_DATE_FORMAT as DF
+
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class recurring_contract(orm.Model):
+    _inherit = "recurring.contract"
+    _order = 'start_date desc'
+
+    def on_change_partner_id(self, cr, uid, ids, partner_id, context=None):

```

Comment:
This can be partially moved in module sponsorship_compassion :
* a regular contract will never need a correspondant, this a feature is only for sponsorships.
* num_pol_ga must stay for all kind of contracts, it is used to identify a sponsorship given a gift bvr_reference.